src/microbe_stage/SpawnSystem.cs

```diff
@@ -230,6 +251,26 @@ private void SpawnEntities(Vector3 playerPosition, Vector3 playerRotation, int e
         if (existing >= maxAliveEntities)
             return;
 
+        // Here we want to check that the player moved to not basically spawn in circle around him.
+        // Solution inspired by gwen is to check if the player
+        // moves out of a square/cycle around his precedent registered position
+        // Not perfect however as going on and off could still break this.
+        float distanceToLastPosition = (playerPosition - lastRecordedPlayerPosition).Length();
+        bool immobilePlayer = distanceToLastPosition < Constants.PLAYER_IMMOBILITY_ZONE_RADIUS;
```

Comment:
Make a `PLAYER_IMMOBILLITY_ZONE_RADIUS_SQUARED` and use `LengthSquared`. This is faster.